app/src/sharedTest/java/org/oppia/android/app/devoptions/DeveloperOptionsFragmentTest.kt

```diff
@@ -0,0 +1,524 @@
+      scrollToPosition(position = 1)
+      verifyItemDisplayedOnDeveloperOptionsListItem(
+        itemPosition = 1,
+        targetView = R.id.view_logs_text_view
+      )
+      verifyTextOnDeveloperOptionsListItemAtPosition(
+        itemPosition = 1,
+        targetViewId = R.id.event_logs_text_view,
+        stringIdToMatch = R.string.developer_options_event_logs
+      )
+    }
+  }
+
+  @Test
+  fun testDeveloperOptionsFragment_overrideAppBehaviorsIsDisplayed() {
+    launch<DeveloperOptionsTestActivity>(
+      createDeveloperOptionsTestActivityIntent(internalProfileId)
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 2)
+      verifyItemDisplayedOnDeveloperOptionsListItem(
+        itemPosition = 2,
+        targetView = R.id.override_app_behaviors_text_view
+      )
+      verifyTextOnDeveloperOptionsListItemAtPosition(
+        itemPosition = 2,
+        targetViewId = R.id.show_all_hints_solution_text_view,
+        stringIdToMatch = R.string.developer_options_show_all_hints_solution
+      )
+      verifyTextOnDeveloperOptionsListItemAtPosition(
+        itemPosition = 2,
+        targetViewId = R.id.force_network_type_text_view,
+        stringIdToMatch = R.string.developer_options_force_network_type
+      )
+      verifyTextOnDeveloperOptionsListItemAtPosition(
+        itemPosition = 2,
+        targetViewId = R.id.force_crash_text_view,
+        stringIdToMatch = R.string.developer_options_force_crash
+      )
+    }
+  }
+
+  @Test
+  fun testDeveloperOptionsFragment_changeConfig_overrideAppBehaviorsIsDisplayed() {
+    launch<DeveloperOptionsTestActivity>(
+      createDeveloperOptionsTestActivityIntent(internalProfileId)
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(isRoot()).perform(orientationLandscape())
+      scrollToPosition(position = 2)
+      verifyItemDisplayedOnDeveloperOptionsListItem(
+        itemPosition = 2,
+        targetView = R.id.override_app_behaviors_text_view
+      )
+      verifyTextOnDeveloperOptionsListItemAtPosition(
+        itemPosition = 2,
+        targetViewId = R.id.show_all_hints_solution_text_view,
+        stringIdToMatch = R.string.developer_options_show_all_hints_solution
+      )
+      verifyTextOnDeveloperOptionsListItemAtPosition(
+        itemPosition = 2,
+        targetViewId = R.id.force_network_type_text_view,
+        stringIdToMatch = R.string.developer_options_force_network_type
+      )
+      verifyTextOnDeveloperOptionsListItemAtPosition(
+        itemPosition = 2,
+        targetViewId = R.id.force_crash_text_view,
+        stringIdToMatch = R.string.developer_options_force_crash
+      )
+    }
+  }
+
+  // TODO(#3397): When the logic to show all hints and solutions is implemented, write a test to
+  //  check for click operation of the 'Show all hints/solution' switch
+  @Test
+  fun testDeveloperOptionsFragment_hintsAndSolutionSwitchIsUncheck() {
```

Comment:
I think the `configChange` case is missing for this one.

Reply:
Actually, this can be possible only after the issue mentioned in the Kdoc is resolved. I think I should update this in the KDoc as well.

Reply:
Done.